src/TNetwork.cpp

```diff
@@ -153,10 +165,14 @@ void TNetwork::TCPServerMain() {
             if (ec) {
                 beammp_errorf("failed to accept: {}", ec.message());
             }
-            ClientSocket.set_option(rcv_timeout_option{ 120000 }); //timeout of 120seconds
+            // Wait for 2 minutes for the client to send us the first packet
+            rcv_timeout_custom_option option(120000);
```

Comment:
You fail to call `option.set_option(...)` here. Did you forget this, or is that how it works?